### PR TITLE
Update masking method to make it safer to calculate cross entropy loss

### PR DESCRIPTION
**Pull request by adamantivm**
Apparently -1e32 is better for masking than -np.inf

```
>>> import torch
>>> import torch.nn.functional as F
>>> x = torch.Tensor([0,1])
>>> y = torch.log(x)
>>> F.cross_entropy(y,x)
tensor(nan)
>>> y
tensor([-inf, 0.])
>>> y[0] = -1e32
>>> F.cross_entropy(y,x)
tensor(-0.)
```

### CODE REVIEW

**Comment by adamantivm** [comment]
ready for review @jonbinney

File: deep_quoridor/src/agents/alphazero/nn_evaluator.py [hunk 1]
```diff
@@ -36,20 +38,20 @@ def evaluate(self, game: Quoridor):
...
-            unmasked_policy_logits = unmasked_policy_logits.cpu().numpy()
-
-            # Set logits of invalid actions to negative infinity to ensure they are never chosen
-            # TODO: Mask entirely in Torch to avoid going to Numpy and back
-            masked_logits = np.full_like(unmasked_policy_logits, -np.inf)
-            masked_logits[valid_action_indices] = unmasked_policy_logits[valid_action_indices]
-
-            policy_masked = F.softmax(torch.from_numpy(masked_logits), dim=-1).cpu().numpy()
+            policy_logits, value = self.network(torch.from_numpy(input_array).float().to(self.device))
+            assert torch.isfinite(policy_logits).all(), "Policy logits contains non-finite values"
```
Comment (jonbinney) [inline deep_quoridor/src/agents/alphazero/nn_evaluator.py:52]: The lines after the assert will never do anything, right?

Comment (jonbinney) [reply deep_quoridor/src/agents/alphazero/nn_evaluator.py:52]: (if we decide that getting -inf out of the network is expected behavior, then it will be nice to have those two lines)

File: deep_quoridor/src/agents/alphazero/nn_evaluator.py [hunk 2]
```diff
@@ -36,20 +38,20 @@ def evaluate(self, game: Quoridor):
 
         self.network.eval()  # Disables dropout
 
+        # Prepare a tensor with True only on INVALID action positions so that we can set those values to
+        # a large negative number below, to discourage choosing them.
```
Comment (jonbinney) [inline deep_quoridor/src/agents/alphazero/nn_evaluator.py:42]: I don't think that this comment quite describes what the following lines do. They're creating a tensor with the valid indices as integers, eg. instead of `[True, False, True, True]` it would contain `[0, 2, 3]` if i understand correctly. This could explain why you had trouble getting ignore_indices to do what you expected.

**Review by jonbinney — APPROVED** [review]
A couple questions, but looks good to merge once those are answered.
